src/components/views/rooms/RoomPreviewBar.js

```diff
@@ -212,13 +220,46 @@ module.exports = React.createClass({
                 break;
             }
             case MessageCase.OtherThreePIDError: {
-                // "Unable to ascertain that the address this invite was sent to matches one associated with your account."
-                // need to show invited by "name", join buttons here?
+                title = _t("Something went wrong with your invite to this room");
+                const joinRule = this._joinRule();
+                const errCodeMessage = _t("%(errcode)s was returned while trying to valide your invite. You could try to pass this information on to a room admin.", {errcode: this.state.threePidFetchError.errcode});
+                switch (joinRule) {
+                    case "invite":
+                        subTitle = [
+                            <p key="subTitle1">{_t("Sadly, you can only join it with a working invite.")}</p>,
+                            <p key="subTitle2">{ errCodeMessage }</p>,
+                        ];
+                        break;
+                    case "public":
+                        subTitle = _t("Luckily, you can still join it because this is a public room.");
```

Comment:
Hmm, you may want to have @nadonomy review the wording here. The "sadly" / "luckily" gives a Riot a certain tone that feels different to other bits of the app. (I don't see this text in Zeplin or the Google doc, but maybe it was already vetted elsewhere.)

Reply:
Ah, looks like they change later.